spec/models/schedule_rule_spec.rb

```diff
@@ -12,6 +15,52 @@
     @rule.should be_valid
   end
 
+  describe ".unavailable_for_date" do
+    context "for an instrument available only from 9 AM to 5 PM" do
+      before(:each) do
+        instrument.schedule_rules.create(attributes_for(:schedule_rule))
+      end
+
+      let(:reservations) { ScheduleRule.unavailable_for_date(instrument, day) }
+
+      shared_examples_for "it generates reservations to cover unavailability" do
+        it "returns two dummy reservations" do
+          expect(reservations.size).to eq(2)
+          reservations.each do |reservation|
+            expect(reservation).to be_kind_of(Reservation)
+            expect(reservation).not_to be_persisted
+          end
+        end
+
+        it "reserves midnight to 9 AM as unavailable" do
+          expect(reservations.first.reserve_start_at)
+            .to eq(day.to_time.change(hour: 0, min: 0))
```

Comment:
I wonder if there's a little bit better approach for this test. This seems to be testing the actual implementation a little too closely. Maybe a string comparison?

Not a big deal if you don't like that as much. I'm not against this current implementation.